Conn tx alloc defer

alwa-nordic opened the pull request.

alwa-nordic committed:
Bluetooth: Host: Delay allocating `bt_conn_tx`

Delay allocating `bt_conn_tx` until we need it.

Signed-off-by: Aleksander Wasaznik <aleksander.wasaznik@nordicsemi.no>

alwa-nordic committed:
Bluetooth: Use `CONFIG_BT_CONN_TX_USER_DATA_SIZE`

Replace hardcoded value `8` with `CONFIG_BT_CONN_TX_USER_DATA_SIZE`,
that is `8` but is going to change.

Signed-off-by: Aleksander Wasaznik <aleksander.wasaznik@nordicsemi.no>

jori-nordic commented on subsys/bluetooth/host/conn.c:
```diff
@@ -683,6 +659,25 @@ static int send_frag(struct bt_conn *conn,
+			tx = conn_tx_alloc();
+			if (!tx) {
+				LOG_DBG("Unable to allocate TX context");
+				return -ENOBUFS;
```
runtime assert (should never fail or block).
static assert to have same n of ctlr bufs as tx contexts. (noop `CONFIG_BT_CONN_TX_MAX`).

alwa-nordic committed:
Bluetooth: Host: conn tx event block on free_tx

For better code isolation, the WOULDBLOCK is not predicted, but instead
informed by the processing code itself.

Signed-off-by: Aleksander Wasaznik <aleksander.wasaznik@nordicsemi.no>

alwa-nordic committed:
wip

alwa-nordic committed:
broken l2cap send_sdu rewrite

alwa-nordic committed:
Rewrite l2cap_chan_le_send_sdu

alwa-nordic committed:
Bluetooth: Host: Remove `l2cap_chan_le_send_sdu` paramter `sent`

Signed-off-by: Aleksander Wasaznik <aleksander.wasaznik@nordicsemi.no>

alwa-nordic committed:
fixup! Bluetooth: Host: conn tx event block on free_tx
